nifi-bootstrap/src/test/java/org/apache/nifi/bootstrap/util/TestSecureNiFiConfigUtil.java

```diff
@@ -0,0 +1,208 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.bootstrap.util;
+
+import org.apache.nifi.properties.NiFiPropertiesLoader;
+import org.apache.nifi.security.util.KeyStoreUtils;
+import org.apache.nifi.security.util.TlsException;
+import org.apache.nifi.util.NiFiProperties;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.util.IPAddress;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.xml.bind.DatatypeConverter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.security.InvalidKeyException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class TestSecureNiFiConfigUtil {
+    private Logger logger = LoggerFactory.getLogger("org.apache.nifi.bootstrap.util.TestSecureNiFiConfigUtil");
+
+    private static final boolean EXPECT_STORES_TO_EXIST = true;
+
+    private String nifiPropertiesFile = "src/test/resources/nifi.properties";
+    private String keystorePath = "src/test/resources/keystore.p12";
+    private String truststorePath = "src/test/resources/truststore.p12";
+
+    private NiFiProperties configureSecureNiFiProperties(String testPropertiesFile) throws IOException {
+        Files.copy(new File(testPropertiesFile).toPath(), new File(nifiPropertiesFile).toPath(), StandardCopyOption.REPLACE_EXISTING);
+        SecureNiFiConfigUtil.configureSecureNiFiProperties(nifiPropertiesFile, logger);
+
+        return new NiFiPropertiesLoader().load(nifiPropertiesFile);
+    }
+
+    private static String getFileHash(String filename) throws NoSuchAlgorithmException, IOException {
+        MessageDigest md = MessageDigest.getInstance("MD5");
```

Comment:
Although this is just for testing, the `MD5` algorithm should be avoided.  Recommend using SHA-256, using Apache Commons `DigestUtils` or one of the other NiFi utility classes for hashing.